modules/kubernetes-addons/cert-manager/data.tf

```diff
@@ -0,0 +1,32 @@
+data "aws_route53_zone" "selected" {
+  for_each = contains(var.domain_names, "*") ? {} : { for domain in toset(var.domain_names) : domain => domain }
```

Comment:
```suggestion
  for_each = { for k,v in toset(var.domain_names) : k => v if !contains(var.domain_names, "*") }
```

Reply:
Ok, then explicit list of domains (zones) rather than magic "*".